api/kv_v2.go

```diff
@@ -238,6 +258,41 @@ func (kv *kvv2) Put(ctx context.Context, secretPath string, data map[string]inte
 	return kvSecret, nil
 }
 
+// PutMetadata can be used to fully replace a subset of metadata fields for a
+// given KV v2 secret.
+//
+// It can also be used to create a new secret with just metadata and no secret data yet.
+//
+// To partially replace the values of these metadata fields, use PatchMetadata.
+func (kv *kvv2) PutMetadata(ctx context.Context, secretPath string, metadata KVMetadataInput) error {
+	pathToWriteTo := fmt.Sprintf("%s/metadata/%s", kv.mountPath, secretPath)
+
+	// convert non-nil values to a map we can pass to Logical
+	md := make(map[string]interface{})
+	casRequiredKey := "cas_required"
+	deleteVersionAfterKey := "delete_version_after"
+	maxVersionsKey := "max_versions"
+	customMetadataKey := "custom_metadata"
+
+	md[customMetadataKey] = metadata.CustomMetadata
+	if metadata.MaxVersions != nil {
+		md[maxVersionsKey] = *(metadata.MaxVersions)
+	}
```

Comment:
If we are keeping the input properties values as pointers, should we cast the nil to a zero value so that it actually replaces the existing one?

Reply:
Oh I just realized--Put needs to be its own method separate from Patch because "patch" is actually a capability verb on a policy. Users without "patch" capability but who have "update" need their own method for Put. 

So maybe casting `nil` on Put to the zero value is the right approach, as you first said.